include/mbedtls/check_config.h

```diff
@@ -376,7 +376,15 @@
     defined(MBEDTLS_SHA224_C) || \
     defined(MBEDTLS_SHA256_C) || \
     defined(MBEDTLS_SHA384_C) || \
-    defined(MBEDTLS_SHA512_C) )
+    defined(MBEDTLS_SHA512_C) || \
+    (defined(MBEDTLS_PSA_CRYPTO_C) && \
```

Comment:
Isn't this falling in the same pitfall as usual? Should it be 
`(!PSA_CRYPTO && ANY_MBEDTLS_SHAXXX) || (PSA_CRYPTO && ANY_PSA_WANT_ALG_SHA_XXX)`?
Or ar we assuming that with PSA_CRYPTO on, MBEDTLS_SHAXXX_C will turn on PSA_WANT_xxx?

Reply:
> Or ar we assuming that with PSA_CRYPTO on, MBEDTLS_SHAXXX_C will turn on PSA_WANT_xxx?

I think we are doing more assuming, we're enforcing that in the second commit of this PR, unless I botched it :)

Reply:
Yeah, I went on a commit by commit basis and this immediately cleared it :) I'll have to switch to batching comments for such PRs.